File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        if(head==null){
            return false;
        }
        HashSet<ListNode>ll=new HashSet<>();
        ListNode temp=head;
        
        while(temp!=null){
            if(ll.contains(temp)){
                return true;
            }
            ll.add(temp);
            temp=temp.next;
        }
        return false;
    }
}

File: 0141-linked-list-cycle/0141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        
        unordered_set<ListNode*>ll;
        
        ListNode *temp=head;
        
        while(temp!=nullptr){
            if(ll.find(temp)!=ll.end()){
                return true;
            }
            ll.insert(temp);
            temp=temp->next;
        }
        return false;
        
    }
};